src/charm.py

```diff
@@ -510,6 +506,35 @@ def _config_force_https(self) -> None:
         )
         process.wait_output()
 
+    def _on_anonymize_user_action(self, event: ActionEvent) -> None:
+        """Anonymize data from a user.
+
+        Args:
+            event: Event triggering the anonymize_admin_user action.
+        """
+        username = event.params["username"]
+        container = self.unit.get_container("discourse")
+        if container.can_connect():
+            process = container.exec(
+                [
+                    "bash",
+                    "-c",
+                    f"./bin/bundle exec rake users:anonymize[{username}]",
+                ],
+                user="discourse",
+                working_dir=DISCOURSE_PATH,
+                environment=self._create_discourse_environment_settings(),
+            )
+            try:
+                process.wait_output()
+                event.set_results({"user": f"{username}"})
+            except ExecError as ex:
+                event.fail(
+                    # Parameter validation errors are printed to stdout
+                    f"Failed to anonymize user with username {username}:"
+                    f"{ex.stdout}"  # type: ignore
```

Comment:
What is this `type: ignore` please? This standard requests a comment explaining the why to be added for any rule disables, is that applicable here? https://github.com/canonical/is-charms-contributing-guide/blob/main/CONTRIBUTING.md#static-code-analysis

Reply:
Added an explanation